dbms/src/Columns/ColumnArray.cpp

```diff
@@ -213,6 +215,36 @@ void ColumnArray::updateHashWithValue(size_t n, SipHash & hash) const
         getData().updateHashWithValue(offset + i, hash);
 }
 
+void ColumnArray::updateWeakHash32(WeakHash32 & hash) const
+{
+    auto s = offsets->size();
+    if (hash.getData().size() != s)
+        throw Exception("Size of WeakHash32 does not match size of column: column size is " + std::to_string(s) +
+                        ", hash size is " + std::to_string(hash.getData().size()), ErrorCodes::LOGICAL_ERROR);
+
+    WeakHash32 internal_hash(data->size());
```

Comment:
Didn't get it. Why don't simply update hashes with all corresponding array elements in chain?

Reply:
Because we don't have method for single array element.
It would be possible if offsets will be added to `updateWeakHash` as the second argument.